fairlearn/metrics/all_metrics.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import numpy as np
+from sklearn.metrics import recall_score
+
+from . import DisparityResult, MetricResult
+
+
+def true_positive_rate(y_actual, y_predict, group_id, sample_weight=None):
+    return metric_by_groups(recall_score, y_actual, y_predict, group_id, sample_weight)
+
+
+def selection_rate(y_actual, y_predict, group_id, sample_weight=None):
+    if sample_weight is not None:
+        raise NotImplementedError("selection_rate and sample_weight")
+
+    def selection_metric(y_a, y_p, sample_weight):
+        return np.mean(y_p)
+
+    return metric_by_groups(selection_metric, y_actual, y_predict, group_id, sample_weight)
+
+
+def metric_by_groups(metric_function, y_actual, y_predict, group_id, sample_weight=None):
+    # TODO: Validate y_actual and y_predict are from {0, 1}
+    # TODO: Validate that group_id are from {0 ... n}
+    result = MetricResult()
+
+    groups = np.unique(group_id)
```

Comment:
group_id indicates a single id, as int or str
This looks more like it's an array?

Reply:
I was following the sklearn convention that the array is singular - it's `sample_weight` not `sample_weights` (I'm not wholly convinced myself....)

Reply:
Hmm I think it's more confusing than anything else. Not convinced this is actually a convention https://stackoverflow.com/questions/395739/do-you-name-your-arrays-plurally-or-singularly

Reply:
'Convention' was restricted to `sklearn` …. and now I look a little further, even that is not consistent. Personally I would make them plural, but I thought in a conversation last week we'd said to make them singular?

Reply:
I may not have been part of that conversation (or maybe I didn't pay attention), because I can't imagine that I would endorse such a resolution :-)

Reply:
I believe it was a somewhat offhand comment from someone; I may have taken it overly to heart.

Reply:
probably better to make it plural if it makes the code more readable, I never knew scikit-learn had that convention, strange

Reply:
we also discussed group_data; either group_ids or group_data is fine with me.

Reply:
`group_data` it is